pkg/dockerregistry/server/prune.go

```diff
@@ -0,0 +1,201 @@
+
+			// We cannot delete the repository at this point, because it would break Enumerate.
+			reposToDelete = append(reposToDelete, repoName)
+
+			return nil
+		} else if err != nil {
+			return fmt.Errorf("failed to get the image stream %s: %v", repoName, err)
+		}
+
+		repository, err := registry.Repository(ctx, named)
+		if err != nil {
+			return err
+		}
+
+		manifestService, err := repository.Manifests(ctx)
+		if err != nil {
+			return err
+		}
+
+		manifestEnumerator, ok := manifestService.(distribution.ManifestEnumerator)
+		if !ok {
+			return fmt.Errorf("unable to convert ManifestService into ManifestEnumerator")
+		}
+
+		err = manifestEnumerator.Enumerate(ctx, func(dgst digest.Digest) error {
+			if _, ok := inuse[string(dgst)]; ok && imageStreamHasManifestDigest(is, dgst) {
+				logger.Debugf("Keeping the manifest link %s@%s", repoName, dgst)
+				return nil
+			}
+
+			if dryRun {
+				logger.Printf("Would delete the manifest link %s@%s", repoName, dgst)
+			} else {
+				logger.Printf("Deleting manifest link: %s@%s", repoName, dgst)
+				err = manifestService.Delete(ctx, dgst)
+				if err != nil {
+					return fmt.Errorf("failed to delete the manifest link %s@%s: %s", repoName, dgst, err)
+				}
+			}
+
+			return nil
+		})
+		if e, ok := err.(driver.PathNotFoundError); ok {
+			logger.Printf("Skipped manifest link pruning for the repository %s: %s", repoName, e)
+		} else if err != nil {
+			return fmt.Errorf("failed to prune manifest links in the repository %s: %s", repoName, err)
+		}
+
+		return nil
+	})
+	if e, ok := err.(driver.PathNotFoundError); ok {
+		logger.Warnf("No repositories found: %s", e)
+		return
+	} else if err != nil {
+		logger.Fatal(err)
+	}
+
+	vacuum := storage.NewVacuum(ctx, storageDriver)
+
+	logger.Debugln("Removing repositories")
+	for _, repoName := range reposToDelete {
+		if dryRun {
+			logger.Printf("Would delete the repository %s", repoName)
+		} else {
+			err = vacuum.RemoveRepository(repoName)
+			if err != nil {
+				logger.Fatal("unable to remove the repository %s: %v", repoName, err)
+			}
+		}
+	}
+
+	logger.Debugln("Processing blobs")
+	blobStatter := registry.BlobStatter()
+	err = registry.Blobs().Enumerate(ctx, func(dgst digest.Digest) error {
+		if imageReference, ok := inuse[string(dgst)]; ok {
+			logger.Debugf("Keeping the blob %s (it belongs to the image %s)", dgst, imageReference)
+			return nil
+		}
+
+		desc, err := blobStatter.Stat(ctx, dgst)
+		if err != nil {
+			return err
+		}
+
+		stats.Blobs++
+		stats.DiskSpace += desc.Size
+
+		if dryRun {
+			logger.Printf("Would delete the blob %s", dgst)
+		} else {
+			err := vacuum.RemoveBlob(string(dgst))
+			if err != nil {
+				return fmt.Errorf("failed to delete the blob %s: %s", dgst, err)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	fmt.Printf("Blobs deleted: %d\n", stats.Blobs)
```

Comment:
"Blobs deleted" / "Blobs would be deleted"
"Disk space freed" / "Disk space would be freed"
Is it better for you? I don't know how to rephrase it so that it begins with the word *would*.